recipe/meta.yaml

```
@@ -0,0 +1,161 @@
+    - patchelf --set-rpath '$ORIGIN/../../torch/lib' $SP_DIR/torchaudio/lib/_torchaudio.so         # [linux]
+    - patchelf --add-rpath '$ORIGIN' $SP_DIR/torchaudio/lib/_torchaudio.so                         # [linux]
+  ignore_run_exports:
+    - pytorch-gpu  # [osx and arm64]
+    - libprotobuf  # [osx and arm64]
+    - libabseil    # [osx and arm64]
+  missing_dso_whitelist:
+    - lib/libstdc++.so.*            # [linux]
+    - lib/libgcc_s.so.*             # [linux]
+    - lib/libgomp.so.*              # [linux]
+    - $RPATH/ld64.so.1              # [linux and s390x]
+    - x86_64-conda-linux-gnu/sysroot/lib64/libm.so.*        # [linux and x64_64]
+    - x86_64-conda-linux-gnu/sysroot/lib64/libc.so.*        # [linux and x64_64]
+    - x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0  # [linux and x64_64]
+    - s390x-conda-linux-gnu/sysroot/lib64/libc.so.*         # [linux and s390x]
+    # contained in ffmpeg
+    - $RPATH/libavutil.*.dylib      # [osx]
+    - $RPATH/libavcodec.*.dylib     # [osx]
+    - $RPATH/libavformat.*.dylib    # [osx]
+    - $RPATH/libavdevice.*.dylib    # [osx]
+    - $RPATH/libavfilter.*.dylib    # [osx]
+    - $RPATH/libavutil.so.*         # [linux]
+    - $RPATH/libc10.so              # [linux]
+    - $RPATH/libavutil.so.*         # [linux]
+    - $RPATH/libavformat.so.*       # [linux]
+    - $RPATH/libavfilter.so.*       # [linux]
+    # libsox shouldn't be activated to begin with
+    - $RPATH/libsox.dylib           # [osx]
+    # in pytorch package
+    - $RPATH/libtorch.dylib         # [osx]
+    - $RPATH/libtorch_cpu.dylib     # [osx]
+    - $RPATH/libtorch_python.dylib  # [osx]
+    - $RPATH/libtorch_cpu.so        # [linux]
+    - $RPATH/libtorch_python.so     # [linux]
+    - $RPATH/libc10.dylib           # [osx]
+    # in protobuf package
+    - $RPATH/libprotobuf.*.dylib    # [osx]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - cmake
+    - make  # [not win]
+    - patch     # [not win]
+    - m2-patch  # [win]
+    - patchelf  # [linux]
+    - protobuf {{ libprotobuf }}
+    - llvm-openmp  # [osx]
+    - intel-openmp  # [not osx and blas_impl == 'mkl']
+  host:
+    - python
+    - setuptools
+    - wheel
+    - pip
+    - pytorch 2.1.0
+    - pybind11
+    - libabseil
+    - ffmpeg 6.1.1
+  run:
+    - python
+    # according to compatibility matrix in
+    # https://github.com/pytorch/audio/blob/v2.1.2/docs/source/installation.rst#compatibility-matrix
+    - pytorch >=2.1.0,<2.2.0
+    # according to optional dependencies
+    # https://github.com/pytorch/audio/blob/v2.1.2/docs/source/installation.rst#optional-dependencies
+    # in this special case we want users of channel sfe140ed to be able to use the load and save functions, therefore
+    # at least ffmpeg (the mightiest of the backends) is specified as a run dependency.
+    - ffmpeg >=4.4,<7
```

```suggestion
    - ffmpeg 6 #
```
We only compile with one version. (we could also rely on the run_exports)

---

fair point, but it should. `6.1.1` in our case.

---

ffmpeg has a relatively stable ABI. They document that they break the ABI on each new major version, see https://www.ffmpeg.org/developer.html#Library-public-interfaces, so technically 6 should be fine (we might also consider adjusting ffmpeg's run_export).

---

pin_compatible? (does it need to be >=6.1.1, <7?)

---

it needs to be `>=6, <7` because of major releases possibly breaking the API.

---

ok but are versions between 6.0.0 and 6.1.1 ok

---

@danpetry see https://github.com/AnacondaRecipes/torchaudio-feedstock/pull/1#discussion_r1681077312. Also see my comment in parenthesis about changing `run_exports`.

---

this was a socratic question... if it's built against 6.1.1, then it should be >=6.1.1 and <7. There's a run_exports now anyway so a hard pin in host and nothing in run is what would be required no?